kernel/src/engine/parquet_stats_skipping.rs

```diff
@@ -0,0 +1,1267 @@
+            let skip_hi = self.partial_cmp_max_stat(&col, val, max_ord, false)?;
+            let skip = skip_lo || skip_hi;
+            Some(skip != inverted)
+        };
+        match op {
+            // Given `col == val`:
+            // skip if `val` cannot equal _any_ value in [min, max], implies
+            // skip if `NOT(val BETWEEN min AND max)` implies
+            // skip if `NOT(min <= val AND val <= max)` implies
+            // skip if `min > val OR max < val`
+            // keep if `NOT(min > val OR max < val)`
+            Equal => min_max_disjunct(Ordering::Greater, Ordering::Less, true),
+            // Given `col != val`:
+            // skip if `val` equals _every_ value in [min, max], implies
+            // skip if `val == min AND val == max` implies
+            // skip if `val <= min AND min <= val AND val <= max AND max <= val` implies
+            // skip if `val <= min AND max <= val` implies
+            // keep if `NOT(val <= min AND max <= val)` implies
+            // keep if `val > min OR max > val` implies
+            // keep if `min < val OR max > val`
+            NotEqual => min_max_disjunct(Ordering::Less, Ordering::Greater, false),
+            // Given `col < val`:
+            // Skip if `val` is not greater than _all_ values in [min, max], implies
+            // Skip if `val <= min AND val <= max` implies
+            // Skip if `val <= min` implies
+            // Keep if `NOT(val <= min)` implies
+            // Keep if `val > min` implies
+            // Keep if `min < val`
+            LessThan => self.partial_cmp_min_stat(&col, val, Ordering::Less, false),
+            // Given `col <= val`:
+            // Skip if `val` is less than _all_ values in [min, max], implies
+            // Skip if `val < min AND val < max` implies
+            // Skip if `val < min` implies
+            // Keep if `NOT(val < min)` implies
+            // Keep if `NOT(min > val)`
+            LessThanOrEqual => self.partial_cmp_min_stat(&col, val, Ordering::Greater, true),
+            // Given `col > val`:
+            // Skip if `val` is not less than _all_ values in [min, max], implies
+            // Skip if `val >= min AND val >= max` implies
+            // Skip if `val >= max` implies
+            // Keep if `NOT(val >= max)` implies
+            // Keep if `NOT(max <= val)` implies
+            // Keep if `max > val`
+            GreaterThan => self.partial_cmp_max_stat(&col, val, Ordering::Greater, false),
+            // Given `col >= val`:
+            // Skip if `val is greater than _every_ value in [min, max], implies
+            // Skip if `val > min AND val > max` implies
+            // Skip if `val > max` implies
+            // Keep if `NOT(val > max)` implies
+            // Keep if `NOT(max < val)`
+            GreaterThanOrEqual => self.partial_cmp_max_stat(&col, val, Ordering::Less, true),
+            _ => None, // unsupported operation
```

Comment:
Can we log that we found an unsupported op

Reply:
Updated them all.